lightly_studio/src/lightly_studio/selection/select_via_db.py

```diff
@@ -62,20 +64,71 @@ def _aggregate_class_distributions(
         annotation_id: j for j, annotation_id in enumerate(target_annotation_ids)
     }
     for i, sample_id in enumerate(input_sample_ids):
-        for annotation in sample_id_to_annotations[sample_id]:
-            label_idx = annotation_id_to_idx.get(annotation.annotation_label_id)
+        for annotation_label_id in sample_id_to_annotation_label_ids[sample_id]:
+            label_idx = annotation_id_to_idx.get(annotation_label_id)
             if label_idx is not None:
                 class_distributions[i, label_idx] += 1
 
     return class_distributions
 
 
+def _process_explicit_target_distribution(
+    session: Session,
+    target_distribution: dict[str, float],
+    annotations: Sequence[AnnotationBaseTable],
+) -> tuple[dict[UUID, float], set[UUID], float]:
+    """Processes the explicit target distribution.
+
+    Args:
+        session: The SQLAlchemy session.
+        target_distribution:
+            A dictionary mapping annotation label names to their target proportions.
+        annotations:
+            A sequence of all annotations to consider for class balancing.
+
+    Returns:
+        A dictionary mapping annotation label IDs to their effective target proportions, the set of
+        unused label IDs and the target value remaining to 1.0.
+
+    Raises:
+        NotImplementedError: If multiple labels with the same name are found.
+        ValueError: If an annotation label name does not exist or if targets sum
+            to less than 1.0 and all classes are used.
+    """
+    label_id_to_target: dict[UUID, float] = {}
+    total_targets = 0.0
+    for label_name, target in target_distribution.items():
+        try:
+            annotation_label = annotation_label_resolver.get_by_label_name(session, label_name)
+        except sqlalchemy.exc.MultipleResultsFound as e:
+            raise NotImplementedError(
+                "Multiple labels with the same name not supported yet."
+            ) from e
+        if annotation_label is None:
+            raise ValueError(f"Annotation label with this name does not exist: {label_name}")
+        label_id_to_target[annotation_label.annotation_label_id] = target
+        total_targets += target
+
+    if 1.0 - total_targets > EPSILON:
+        all_label_ids = {a.annotation_label_id for a in annotations}
+        unused_label_ids = all_label_ids - set(label_id_to_target.keys())
+
+        unused_count = len(unused_label_ids)
+        if unused_count == 0:
+            raise ValueError(
+                f"Targets sum to less than 1.0 and all classes are used: {total_targets}"
+            )
+        return label_id_to_target, unused_label_ids, 1.0 - total_targets
```

Comment:
We must use the other class even if total_targets >= 1. In that case set the remainder to 0.0. This is necessary to enforce that nothing of the other classes is selected instead of not caring whether they are selected or not. 
I tried it out live on your branch.

```suggestion
    all_label_ids = {a.annotation_label_id for a in annotations}
    unused_label_ids = all_label_ids - set(label_id_to_target.keys())

    unused_count = len(unused_label_ids)
    if unused_count == 0:
        raise ValueError(
            f"Targets sum to less than 1.0 and all classes are used: {total_targets}"
        )
    return label_id_to_target, unused_label_ids, max(1.0 - total_targets, 0)
```

Reply:
Done, also added a test.